### PR TITLE
[DR-2986] Require Load Tag on Bulk Load

### PR DESCRIPTION
**Pull request by samanehsan**
https://broadworkbench.atlassian.net/browse/DR-2986

If a user leaves the load tag blank when performing a bulk ingest request, TDR will generate a load tag based on the flight id. However, if a user needs to retry the same ingest request several times (because some of the files failed to ingest), it is easier if they reuse a single load tag that they define instead of finding the TDR-generated load tag for each ingest request.

**Changes:**
Throw an error if the load tag is blank in the `BulkLoadRequestModel` or the `BulkLoadArrayRequestModel` (before calling the flight).

### CODE REVIEW

**Review by snf2ye — APPROVED** [review]

File: src/main/java/bio/terra/service/filedata/FileService.java [hunk 1]
```diff
@@ -113,6 +115,10 @@ public String ingestFile(
 
   public String ingestBulkFile(
       String datasetId, BulkLoadRequestModel loadModel, AuthenticatedUserRequest userReq) {
+    if (loadModel.isBulkMode() && StringUtils.isEmpty(loadModel.getLoadTag())) {
```
Comment (nmalfroy) [inline src/main/java/bio/terra/service/filedata/FileService.java]: I think this logic probably should live in the `bio.terra.service.dataset.IngestRequestValidator` class?

File: src/main/java/bio/terra/service/filedata/FileService.java [hunk 2]
```diff
@@ -143,6 +149,9 @@ public String ingestBulkFile(
 
   public String ingestBulkFileArray(
       String datasetId, BulkLoadArrayRequestModel loadArray, AuthenticatedUserRequest userReq) {
+    if (loadArray.isBulkMode() && StringUtils.isEmpty(loadArray.getLoadTag())) {
```
Comment (nmalfroy) [inline src/main/java/bio/terra/service/filedata/FileService.java]: Same with this logic

**Review by nmalfroy — CHANGES_REQUESTED** [review]
Logic looks good! Just wondering if the validation is happening in the right place

**Review by nmalfroy — APPROVED** [review]
thanks for making the change!

**Pull request merged**